python_modules/dagster/dagster/_core/definitions/multi_dimensional_partitions.py

```diff
@@ -0,0 +1,76 @@
+from typing import List, Mapping, NamedTuple, Optional, Sequence, Union
+
+import dagster._check as check
+from dagster._serdes import whitelist_for_serdes
+from dagster._serdes.serdes import deserialize_as, serialize_dagster_namedtuple
+
+
+@whitelist_for_serdes
+class PartitionDimensionKey(
+    NamedTuple(
+        "_PartitionDimensionKey",
+        [
+            ("dimension_name", str),
+            ("partition_key", str),
+        ],
+    )
+):
+    """
+    Serializable representation of a single dimension of a multi-dimensional partition key.
+    """
+
+    def __new__(cls, dimension_name: str, partition_key: str):
+        return super(PartitionDimensionKey, cls).__new__(
+            cls,
+            dimension_name=check.str_param(dimension_name, "dimension_name"),
+            partition_key=check.str_param(partition_key, "partition_key"),
+        )
+
+
+@whitelist_for_serdes
+class MultiDimensionalPartitionKey(
+    NamedTuple("_MultiDimensionalPartitionKey", [("dimension_keys", List[PartitionDimensionKey])])
```

Comment:
Are there particular reasons not to use a `Mapping[str, str]`?

Reply:
Currently we serialize this value (and mappings cannot be serialized).

But this also enforces that the dimension keys will be ordered by dimension name, to allow for equivalence checks and to ensure a consistent ordering of the stringified partition key (e.g. always displaying "US|2022-10-19" instead of "2022-10-19|US")

Reply:
Got it - makes sense